lib/components/narrative/default/default-itinerary.js

```diff
@@ -129,66 +174,83 @@ export default class DefaultItinerary extends NarrativeItinerary {
+            onClick={this._onHeaderClick}
+          >
+            <div
+              className='title'>
+              {getItineraryDescription(itinerary)}
+            </div>
+            <ul className='list-unstyled itinerary-attributes'>
+              {ITINERARY_ATTRIBUTES
+                .sort((a, b) => {
+                  const aSelected = this._isSortingOnAttribute(a)
+                  const bSelected = this._isSortingOnAttribute(b)
+                  if (aSelected) return -1
+                  if (bSelected) return 1
+                  else return a.order - b.order
+                })
+                .map(attribute => {
+                  const isSelected = this._isSortingOnAttribute(attribute)
+                  const options = attribute.id === 'arrivalTime' ? timeOptions : {}
+                  if (isSelected) {
+                    options.isSelected = true
+                    options.selection = this.props.sort.type
+                  }
+                  options.LegIcon = LegIcon
+                  options.timeFormat = this.props.use24HourFormat ? 'H:mm' : 'h:mm a'
+                  return (
+                    <li key={attribute.id} className={`${attribute.id}${isSelected ? ' main' : ''}`}>
+                      {attribute.render(itinerary, options)}
+                    </li>
+                  )
+                })
+              }
+            </ul>
+            <ItinerarySummary itinerary={itinerary} LegIcon={LegIcon} />
+            {(active && !expanded) &&
+              <small style={{clear: 'both', textAlign: 'center', display: 'block', color: 'grey'}}>
+                <FormattedMessage id='clickDetails' />
+              </small>
             }
-          </ul>
-          <ItinerarySummary itinerary={itinerary} LegIcon={LegIcon} />
-          {(active && !expanded) &&
-            <small style={{clear: 'both', textAlign: 'center', display: 'block', color: 'grey'}}>
-              click to view details
-            </small>
+          </button>
+          {(active && expanded) &&
+            <>
+              {showRealtimeAnnotation && <SimpleRealtimeAnnotation />}
+              <ItineraryBody timeOptions={timeOptions} itinerary={itinerary} LegIcon={LegIcon} setActiveLeg={setActiveLeg} />
+            </>
           }
-        </button>
-        {(active && expanded) &&
-          <>
-            {showRealtimeAnnotation && <SimpleRealtimeAnnotation />}
-            <ItineraryBody timeOptions={timeOptions} itinerary={itinerary} LegIcon={LegIcon} setActiveLeg={setActiveLeg} />
-          </>
-        }
-      </div>
+        </div>
+      </IntlProvider>
     )
   }
 }
+
+const mapStateToProps = (state, ownProps) => {
+  return {
+    locale: getUrlParams().locale || 'en-US',
+    userLanguage: state.user.loggedInUser == null ? 'en-US' : state.user.loggedInUser.language,
+    use24HourFormat: state.user.loggedInUser == null ? false : state.user.loggedInUser.use24HourFormat
```

Comment:
These all deserve a little more thought on how they are derived. I think a good place a lot of this could live is somewhere in the `ui` object in the redux state. Even if a user has a default language or time format they might still want to change it to some other language.